src/emulator/storage/files.ts

```diff
@@ -40,12 +46,29 @@ export class StoredFile {
   }
 }
 
+/**  Parsed request object for {@link StorageLayer#authenticateUser}. */
+export type CreateSignedUrl = {
+  bucketId: string;
+  decodedObjectId: string;
+  authorization?: string;
+  originalUrl: string;
+  ttlSeconds: number;
+};
+
+export type SignedUrlResponse = {
+  signed_url: string;
+};
+
 /**  Parsed request object for {@link StorageLayer#getObject}. */
 export type GetObjectRequest = {
   bucketId: string;
   decodedObjectId: string;
+  url?: string;
   authorization?: string;
   downloadToken?: string;
+  urlSignature?: string;
+  urlUsableSeconds?: string | undefined;
```

Comment:
The `?` operator already marks the field as optional, so `| undefined` is redundant